client/my-sites/checkout/checkout/domain-details-form.jsx

```diff
@@ -380,78 +385,40 @@ export class DomainDetailsForm extends PureComponent {
 		);
 	}
 
-	renderAddressFields() {
-		return (
-			<div>
-				<Input
-					label={ this.props.translate( 'Address' ) }
-					maxLength={ 40 }
-					{ ...this.getFieldProps( 'address-1' ) }
-				/>
-
-				<HiddenInput
-					label={ this.props.translate( 'Address Line 2' ) }
-					text={ this.props.translate( '+ Add Address Line 2' ) }
-					maxLength={ 40 }
-					{ ...this.getFieldProps( 'address-2' ) }
-				/>
-			</div>
-		);
-	}
-
-	renderCityField() {
-		return <Input label={ this.props.translate( 'City' ) } { ...this.getFieldProps( 'city' ) } />;
-	}
-
-	renderStateField() {
-		const countryCode = formState.getFieldValue( this.state.form, 'countryCode' );
-
-		return (
-			<StateSelect
-				label={ this.props.translate( 'State' ) }
-				countryCode={ countryCode }
-				{ ...this.getFieldProps( 'state' ) }
-			/>
-		);
-	}
-
-	renderPostalCodeField() {
-		return (
-			<Input
-				label={ this.props.translate( 'Postal Code' ) }
-				{ ...this.getFieldProps( 'postal-code' ) }
-			/>
-		);
-	}
-
-	renderCountryDependentAddressFields( needsOnlyGoogleAppsDetails ) {
+	renderDomainContactDetailsFields() {
+		const { contactDetails } = this.props,
+			countryCode = ( contactDetails || {} ).countryCode;
 		return (
-			<div className="checkout__domain-details-country-dependent-address-fields">
-				{ ! needsOnlyGoogleAppsDetails && this.renderAddressFields() }
-				{ ! needsOnlyGoogleAppsDetails && this.renderCityField() }
-				{ ! needsOnlyGoogleAppsDetails && this.renderStateField() }
-				{ this.renderPostalCodeField() }
+			<div className="checkout__domain-contact-details-fields">
+				{ this.renderOrganizationField() }
+				{ this.renderEmailField() }
+				{ this.renderPhoneField() }
+				{ this.renderFaxField() }
+				{ countryCode && (
+					<RegionAddressFieldsets
+						getFieldProps={ this.getFieldProps }
+						countryCode={ countryCode }
+						canAutoFocusAddressField={ this.canAutoFocusAddressField }
```

Comment:
"should" seems closer than "can" to what we mean here (the fieldset is always capable of autofocusing, and we're asking it not to).

Reply:
Ah, yes. I was using 'can' in the sense of 'may'. As in _Go ahead, old chap. Let's see that auto focus now._ :)